### PR TITLE
Add descriptions for plugins

### PR DESCRIPTION
**Pull request by djyasin**
Add descriptions for plugins

### CODE REVIEW

File: src/awx_plugins/inventory/plugins.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # FIXME: the following violations must be addressed gradually and unignored
-# mypy: disable-error-code="assignment, no-untyped-call, no-untyped-def, var-annotated"
+# mypy: disable-error-code='assignment, no-untyped-call, no-untyped-def, var-annotated'
```
Comment (webknjaz) [inline src/awx_plugins/inventory/plugins.py]: I'm assuming a misconfigured code editor made a bunch of invasive changes, eh?

Comment (djyasin) [reply src/awx_plugins/inventory/plugins.py]: I ran black on this before I committed and it reformatted everything- I am going to pull out those changes today!

File: src/awx_plugins/inventory/plugins.py [hunk 2]
```diff
@@ -18,6 +18,8 @@ class PluginFileInjector:
     plugin_name = None  # Ansible core name used to reference plugin
     # base injector should be one of None, "managed", or "template"
     # this dictates which logic to borrow from playbook injectors
+    plugin_description = None
+    # this field is for the description of the injector
```
Comment (webknjaz) [inline src/awx_plugins/inventory/plugins.py]: In general, comments should be on the same line if possible:
```suggestion
    plugin_description = None  # this field holds the injector description
```
This is because over time, the comment line and the code line may drift away from each other by accident. And the comments will end up being in weird places.

**Review by webknjaz — DISMISSED** [review]
Does this still need to remain a draft?

P.S. Note that all of the revert+fixup+wip commit titles will end up on `devel`.

File: src/awx_plugins/inventory/plugins.py (hunk 1 above)
Comment (webknjaz) [reply src/awx_plugins/inventory/plugins.py]: Why didn't you run the automation that is integrated into the repo instead?

Comment (djyasin) [reply src/awx_plugins/inventory/plugins.py]: It was just force of habit, I will absolutely do that moving forward!

Comment (webknjaz) [reply src/awx_plugins/inventory/plugins.py]: Totally understandable :)
I was just configuring my editor for the new laptop the other day and the first thing I did was figuring out how to make it non-invasive because the framework I'm on happens to have formatting on save which is a harmful assumption to have universally in context of arbitrary projects I might want to open..

File: src/awx_plugins/inventory/plugins.py [hunk 3]
```diff
@@ -64,6 +69,9 @@ def inventory_as_dict(self, inventory_update, private_data_dir):
             }.{
                 self.plugin_name
             }""" if self.use_fqcn else self.plugin_name
+            {
+                self.plugin_description,
+            }
```
Comment (webknjaz) [inline src/awx_plugins/inventory/plugins.py]:
```suggestion
```

File: src/awx_plugins/inventory/plugins.py [hunk 4]
```diff
@@ -30,7 +31,11 @@ class PluginFileInjector:
     def get_proper_name(cls) -> str | None:
```
Comment (webknjaz) [inline src/awx_plugins/inventory/plugins.py]: If this is used in contexts that expect an FQCN, it'll break things externally.

Comment (djyasin) [reply src/awx_plugins/inventory/plugins.py]: Thank you!

Comment (webknjaz) [reply src/awx_plugins/inventory/plugins.py]: Why did you mark it as resolved, though?

File: src/awx_plugins/inventory/plugins.py [hunk 5]
```diff
@@ -30,7 +31,15 @@ class PluginFileInjector:
...
         if cls.plugin_name is None:
             return None
-        return f'{cls.namespace}.{cls.collection}.{cls.plugin_name}'
+        return f'{
```
Comment (webknjaz) [inline src/awx_plugins/inventory/plugins.py]: If you still want this change for some reason, given that it's likely breaking, it's necessary to fix the syntax and make it a multiline string. Otherwise, revert this change to the working state.

File: src/awx_plugins/inventory/plugins.py (hunk 4 above)
Comment (djyasin) [reply src/awx_plugins/inventory/plugins.py]: My mistake- I am so sorry! Should be fixed now :)

File: src/awx_plugins/inventory/plugins.py (hunk 5 above)
Comment (djyasin) [reply src/awx_plugins/inventory/plugins.py]: I have revereted this to its working state. Thank you again!

**Comment by webknjaz** [comment]
@djyasin with all the changes back and forth, Codecov seems to be confused. Could you rebase to see if that'll fix the problem?

**Comment by djyasin** [comment]
> @djyasin with all the changes back and forth, Codecov seems to be confused. Could you rebase to see if that'll fix the problem?

I just  rebased and squashed, let me know if that looks better!

**Comment by webknjaz** [comment]
@djyasin perfect! Codecov no longer reports 0% patch coverage. The commit description could be cleaned up, but that's not a hard blocker. So let's not delay merging any further.

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
@djyasin *post*-merge instructions:
1. Click `Run workflow` @ https://github.com/ansible/awx-plugins/actions/workflows/ci-cd.yml, entering `0.0.1a7` precisely in the version field. Don't enter anything else, and submit the form.
2. Wait until the newly created workflow run completes tests and pauses, then approve releasing on the workflow run summary page.
3. Wait until the workflow run completes fully.
4. Make sure awx gets the new version. Make a PR bumping the dependency over there if necessary.
